system/Common.php

```diff
@@ -1264,3 +1264,24 @@ function trait_uses_recursive($trait)
         return $traits;
     }
 }
+
+if (! function_exists('request_is')) {
+    /**
+     * Check if a request meets the desired type
+     *
+     * @param string $type
+     *
+     * @return boolean
+     */
+    function request_is(string $type): bool
+    {
+        $request = Services::request(null, true);
+
+        if ($type === 'ajax')
```

Comment:
I think that this "muddies" the function a bit. "AJAX" is not a valid HTTP method, and indeed any method may be AJAX so this does not allow checking for POST AJAX versus GET AJAX, etc. I would be in favor of dropping this since the following one-liner will work anywhere in the framework to cover this use:
```php
service('request')->isAJAX();
```